runelite-client/src/main/java/net/runelite/client/plugins/config/SearchablePlugin.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2020, Jack Hodkinson <https://github.com/jrhodkinson>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.config;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import java.util.function.Supplier;
+import lombok.Getter;
+
+public class SearchablePlugin<T>
+{
+
+	private final T plugin;
+	@Getter
+	private final String name;
+	@Getter
+	private final List<String> keywords;
+	private final Supplier<Boolean> isPinnedSupplier;
+
+	@VisibleForTesting
+	SearchablePlugin(T plugin, String name, List<String> keywords, Supplier<Boolean> isPinnedSupplier)
+	{
+		this.plugin = plugin;
+		this.name = name;
+		this.keywords = keywords;
+		this.isPinnedSupplier = isPinnedSupplier;
+	}
+
+	public static SearchablePlugin<PluginListItem> from(PluginListItem pluginListItem)
```

Comment:
These should be `to` methods on the PluginListItem/PHP.PluginItem and those classes made private again.

It would probably better for this to become an interface and make those classes implement it

Reply:
Went for the interface in the end. One wart is that `getName` already exists on one of the implementers so I had to call the method `getSearchableName`, but it's still better than the original proposal